packages/frontend/src/hooks/useRefreshServer.tsx

```diff
@@ -1,27 +1,57 @@
-import { ApiError, ApiRefreshResults } from 'common';
-import { useMutation, useQueryClient } from 'react-query';
+import { ApiError } from 'common';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
 import { useParams } from 'react-router-dom';
 import { lightdashApi } from '../api';
+import { useApp } from '../providers/AppProvider';
 import useQueryError from './useQueryError';
 
-const refresh = async (projectUuid: string) => {
-    await lightdashApi<ApiRefreshResults>({
+const refresh = async (projectUuid: string) =>
+    lightdashApi<any>({
         method: 'POST',
         url: `/projects/${projectUuid}/refresh`,
         body: undefined,
     });
+
+const getJob = async (jobUuid: string) =>
+    lightdashApi<any>({
+        method: 'GET',
+        url: `/jobs/${jobUuid}`,
+        body: undefined,
+    });
+
+export const useGetRefreshData = (jobId: string | undefined) => {
+    const { projectUuid } = useParams<{ projectUuid: string }>();
+    const { showToastSuccess } = useApp();
+    const queryClient = useQueryClient();
+    const setErrorResponse = useQueryError();
+    return useQuery<any, ApiError>({
+        queryKey: ['refresh', projectUuid],
+        queryFn: () => getJob(jobId || ''),
+        enabled: jobId !== undefined,
```

Comment:
similar to below can we make this dependent on the job results? So if the job is in progress, `enable=true` otherwise if the job is completed/error then `enabled=false` - that way the user of this hook doesn't need to manually disable fetching by setting the `jobId` to `undefined`